modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -135,7 +135,13 @@ const EditorUpload = function (editor: Editor): EditorUpload {
             blobCache.removeByUri(image.src);
             replaceImageUriInView(image, uploadInfo.url);
           } else if (uploadInfo.error) {
-            ErrorReporter.uploadError(editor, uploadInfo.error);
+            if (uploadInfo.error.options.remove) {
+              editor.undoManager.transact(() => {
+                editor.dom.remove(image);
```

Comment:
You'll need to remove the blob from the blob cache here as well. So basically, you'll just need to call:
```
blobCache.removeByUri(image.src);
```

Reply:
This has rtc implications as well we need to override this remove and do that on the model. It's also a bit weird to let the handler choose wouldn't it be better to just have a setting instead would the user need to remove case by case basis on images seems a bit overkill.

Reply:
It's what was requested both on GitHub and Zendesk, so there is a use case for it. Most of the cases normally involve them trying to validate the file and wanting it to explicitly be removed, rather than just be removed because of a backend service failure.